### DISCUSSION TITLE
how to get all query strings array in +server.js ?

### DISCUSSION THREAD

**Discussion by magic-thomas** [sveltejs/kit#8634] [Q&A] [answered]
in './src/routes/api/videos/+server.js ' 

```js
/** @type {import('./$types').RequestHandler} */
export async function GET( { request, response }) {
    console.log(request.url.searchParams.get('video_key') )
}
```

I access this url 
http://localhost:5173/api/videos?video_key=100

But it give me error
```sh
TypeError: Cannot read properties of undefined (reading 'get')
```

How can I get all query string arrays in +server.js ?

**Comment by teemingc** [comment, marked as answer]
The official `Request` interface only provides `request.url` as a string.

https://developer.mozilla.org/en-US/docs/Web/API/Request/url

However, you can access the `URL` interface from the arguments instead such as
```diff
/** @type {import('./$types').RequestHandler} */
- export async function GET( { request, response }) {
+ export async function GET( { url }) {
-     console.log(request.url.searchParams.get('video_key') )
+     console.log(url.searchParams.get('video_key') )
}
```
